.github/workflows/dotnet-build-and-test.yml

```diff
@@ -127,7 +127,7 @@ jobs:
           targetdir: "./TestResults/Reports"
           reporttypes: "JsonSummary"
           # Report for production packages only
-          assemblyfilters: "+Microsoft.SemanticKernel.Abstractions;+Microsoft.SemanticKernel.Core;+Microsoft.SemanticKernel.PromptTemplates.Handlebars;+Microsoft.SemanticKernel.Connectors.OpenAI;+Microsoft.SemanticKernel.Yaml;"
+          assemblyfilters: "+Microsoft.SemanticKernel.Abstractions;+Microsoft.SemanticKernel.Core;+Microsoft.SemanticKernel.PromptTemplates.Handlebars;+Microsoft.SemanticKernel.Connectors.OpenAI;+Microsoft.SemanticKernel.Yaml;+Microsoft.SemanticKernel.Agents.Abstractions;+Microsoft.SemanticKernel.Agents.Core;+Microsoft.SemanticKernel.Agents.OpenAI"
```

Comment:
As comment says, we are including code coverage check for production packages only. If we don't expect huge changes in these assemblies in the future, then we can include them and update this comment. If you expect big changes and more experimentation in the future, this step will slow the delivery process. It's not a bad thing, it's just for your information.

Another thing, since this PR does not include changes in .NET, the tests and coverage check weren't invoked. It would be nice to see test code coverage run to verify that all packages satisfy 80% coverage threshold.

Reply:
Runtime doesn't look affected much, but I need to increase coverage (I guess I was measuring the `AgentSyntaxExamples` coverage along with UT locally):

![image](https://github.com/microsoft/semantic-kernel/assets/66376200/8fe16e3b-8b12-40ca-a7a0-9c62271eabd5)

Reply:
> I do feel maintaining normal dev-metrics on these is important as we do plan on graduating these and they are intended to become an important component in the SK eco-system.

Yes, totally agree!

> Runtime doesn't look affected much, but I need to increase coverage

It's good that we caught that here, because if we would merge this PR as it is, it would break other PR pipelines with low coverage error.

Reply:
Agree - This is what I was seeing locally.  So glad we got these numbers.  Thanks for suggestion.

![image](https://github.com/microsoft/semantic-kernel/assets/66376200/815928ef-527f-407d-baf4-cfcf0ff9736e)

Reply:
Unit-Tests Only:

![image](https://github.com/microsoft/semantic-kernel/assets/66376200/4ae04151-7910-444a-9ac2-e2de5a13a7e7)